addons/controllers/packageinstallstatus_controller_test.go

```diff
@@ -0,0 +1,221 @@
+		mngAntreaObjKey         = client.ObjectKey{Namespace: constants.TKGSystemNS, Name: antreaPkgShortName}
+		wlcAntreaObjKey         = client.ObjectKey{Namespace: constants.TKGSystemNS, Name: util.GeneratePackageInstallName(clusterNameWlc, antreaPkgRefName)}
+		wlcKappObjKey           = client.ObjectKey{Namespace: clusterNamespaceWlc, Name: util.GeneratePackageInstallName(clusterNameWlc, kappPkgRefName)}
+	)
+
+	JustBeforeEach(func() {
+		By("Creating resources")
+		f, err := os.Open(clusterResourceFilePath)
+		Expect(err).ShouldNot(HaveOccurred())
+		defer f.Close()
+		Expect(testutil.CreateResources(f, cfg, dynamicClient)).Should(Succeed())
+
+		By("Installing antrea into management cluster")
+		testInstallPackage(clusterNameMng, antreaPkg, antreaPkgShortName, constants.TKGSystemNS)
+		testUpdatePkgInstallStatus(mngAntreaObjKey, kappctrlv1alpha1.ReconcileSucceeded)
+
+		By("Creating kubeconfig for management cluster")
+		Expect(testutil.CreateKubeconfigSecret(cfg, clusterNameMng, clusterNamespaceMng, k8sClient)).Should(Succeed())
+
+		By("Creating kubeconfig for workload cluster")
+		Expect(testutil.CreateKubeconfigSecret(cfg, clusterNameWlc, clusterNamespaceWlc, k8sClient)).Should(Succeed())
+	})
+
+	AfterEach(func() {
+		By("Deleting resources")
+		f, err := os.Open(clusterResourceFilePath)
+		Expect(err).ShouldNot(HaveOccurred())
+		defer f.Close()
+		// Best effort resource deletion
+		_ = testutil.DeleteResources(f, cfg, dynamicClient, true)
+	})
+
+	Context("reconcile PackageInstallStatus", func() {
+		BeforeEach(func() {
+			clusterResourceFilePath = "testdata/test-packageinstallstatus.yaml"
+		})
+
+		It("Should reconcile PackageInstallStatus on management & workload clusters", func() {
+
+			By("verifying management cluster is created properly")
+			mngCluster := &clusterapiv1beta1.Cluster{}
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: clusterNamespaceMng, Name: clusterNameMng}, mngCluster)).Should(Succeed())
+			// update management cluster status phase to 'Provisioned' as it is required for ClusterBootstrap reconciler
+			mngCluster.Status.Phase = string(clusterapiv1beta1.ClusterPhaseProvisioned)
+			Expect(k8sClient.Status().Update(ctx, mngCluster)).Should(Succeed())
+
+			By("verifying workload cluster is created properly")
+			wlcCluster := &clusterapiv1beta1.Cluster{}
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: clusterNamespaceWlc, Name: clusterNameWlc}, wlcCluster)).Should(Succeed())
+			// update workload cluster status phase to 'Provisioned' as it is required for ClusterBootstrap reconciler
+			wlcCluster.Status.Phase = string(clusterapiv1beta1.ClusterPhaseProvisioned)
+			Expect(k8sClient.Status().Update(ctx, wlcCluster)).Should(Succeed())
+
+			By("verifying management and workload clusters' ClusterBootstrap has been created")
+			mngClusterBootstrap := testClusterBootstrapGet(client.ObjectKeyFromObject(mngCluster))
+			Expect(mngClusterBootstrap).ShouldNot(BeNil())
```

Comment:
question: would it be better put L95's assertion into `testClusterBootstrapGet` ?

Reply:
same to the following as well.